### DISCUSSION TITLE
Introducing FIREpyDAQ

### DISCUSSION THREAD

**Discussion by dushyant-fire** [ni/nidaqmx-python#635] [Show and tell]
Hello all,

This python API has been a great source of inspiration to streamline some of the data acquisition that I conduct regularly for some of the experiments (related to fire research). Thank you for maintaining this package. 

I was working on creating a graphical user interface using `PySide6` to facilitate continuous data acquisition for multiple channels at relatively low frequency (<10Hz) from NI devices and some other devices. While working on it, it led to the creation of a python package which I would like to briefly introduce here. 

The package is called [`firepydaq`](https://github.com/ulfsri/firepydaq) which uses this `nidaqmx` python package to setup continuous acquisition tasks during experiments using a user-defined .csv config file (for voltage, thermocouple, and current). Optionally, it can read a user-defined formulae file to post-process live data in a parallel process, that does not interfere with the acquisition process, and display the processed data in a local web-hosted dashboard (using `dash`). It also has the ability to read data from some other devices regularly used during such experiments (Alicat and Thorlabs devices). 

I was hoping to share this information in case someone is interested to contribute to this package or have any feedback or suggestions. Happy to hear any thoughts from this community.

Thank you,
Dushyant

**Comment by zhindes** [comment]
Very slick, thanks for sharing! I only took a cursory glance, but I'm quite impressed. I'm very excited to see examples of people using our API for stuff like this. I'll dig in more in the future, for sure :)
